perses/tests/test_relative.py

```diff
@@ -1047,3 +1047,112 @@ def test_RESTCapableHybridTopologyFactory_energies_GPU():
         # Run MD validation test
         run_RESTCapableHybridTopologyFactory_energies(test_name, phase, use_point_energies=False)
 
+def run_unsampled_endstate_energies(test_name, use_point_energies=True, use_md_energies=False):
+    """
+    Check that the energies of the unsampled endstate hybrid systems generated by dispersed/utils.py/create_endstates_from_real_systems()
+    match the energies of the original hybrid system.
+
+    Checks the energies using validate_unsampled_endstates_point() and/or validate_unsampled_endstates_md() -- first for
+    RESTCapableHybridTopologyFactory and then for HybridTopologyFactory
+
+    Parameters
+    ----------
+    test_name : str
+        Name of the test system. Currently supports: 'ala-dipeptide', 'barstar'.
+    use_point_energies : boolean, default True
+        Whether to run the point energy test for energy validation.
+    use_md_energies : boolean, default False
+        Whether to run the MD energy test for energy validation.
+
+    """
+
+    from perses.dispersed.utils import create_endstates_from_real_systems
+    from perses.tests.test_topology_proposal import generate_atp, generate_dipeptide_top_pos_sys
+    from perses.app.relative_point_mutation_setup import  PointMutationExecutor
+    from perses.tests.utils import validate_unsampled_endstates_point, validate_unsampled_endstates_md
+
+    if test_name == 'ala-dipeptide':
+        # Generate topology, system, systems, system generator for ALA dipeptide in solvent
+        atp, system_generator = generate_atp(phase='solvent')
+
+        # Generate hybrid factory for dipeptide mutation ALA -> THR
+        htfs = []
+        for generate_rest_factory in [True, False]:
+            htf = generate_dipeptide_top_pos_sys(atp.topology,
+                                                 'THR',
+                                                 atp.system,
+                                                 atp.positions,
+                                                 system_generator,
+                                                 conduct_htf_prop=True,
+                                                 generate_rest_capable_hybrid_topology_factory=generate_rest_factory,
+                                                 validate_endstate_energy=False)
+            htfs.append(htf)
+
+    elif test_name == 'barstar':
```

Comment:
What about the case where the test name is neither `barstar` or `ala-dipeptide`? Since right now if the test name is niether, it will just keep going after the if, elif block.

Reply:
I addressed this concern in the most recent commit.